updated ext docs (#1485)
kyma-project/busola: docs/extensibility/resources.md

# Config Map for resource-based extensions

## Overview

> This document describes the required Config Map sections, you need to configure in order to handle your CRD UI page.

Reviewer: ```suggestion
This document describes the required Config Map sections that you need to configure in order to handle your CRD UI page.
```

All sections can be provided as either JSON or YAML.

## resource section

The `resource` section is required and contains basic information about the resource. For example, `kind` and API details.

- **kind** - _[required]_ Kubernetes kind of the resource.
- **group** - _[required]_ API group used for all requests.
- **version** - _[required]_ API version used for all requests.
- **scope** - either `namespace` or `cluster`. Defaults to `cluster`.

> - **path** - path fragment for this resource used in the url. Defaults to pluralized lowercase `kind`. Used to provide an alternative url to avoid conflicts with other resources.

Reviewer: ```suggestion
- **path** - path fragment for this resource used in the URL. Defaults to pluralized lowercase **kind**. Used to provide an alternative URL to avoid conflicts with other resources.
```

}
```

## schema section

The `schema` section contains the JSON-schema definition of the resource. In most cases this is copied verbatim from the CRD. The schema is the basis for generating the create/edit forms and the resultant resource yaml using [ui-schema](https://ui-schema.bemit.codes/).

## form section

> The `form` section contains a list of objects that define which fields you must include in the final form. All given fields are placed in the advanced form by default. It's possible to add a field to the simple form by providing the `simple: true` flag, or removing it from the advanced form by providing the `advanced: false` flag.

Reviewer: ```suggestion
The `form` section contains a list of objects that define which fields you must include in the final form. All given fields are placed in the advanced form by default. It's possible to add a field to the simple form by providing the `simple: true` flag. You can also remove it from the advanced form by providing the `advanced: false` flag.
```

Reviewer: Makes the sentence shorter and more digestible

If you target elements of an array rather that the array itself, you can use `items[]` notation.

### Item parameters

> - **path** - _[required]_ path to the property that you want to display in the form. In case of an array, the array index is omitted. For example, if `spec.items` is an array and you want to display `name` for each items, the path is `spec.items.name`.

Reviewer: ```suggestion
- **path** - _[required]_ path to the property that you want to display in the form. In the case of an array, the array index is omitted. For example, if `spec.items` is an array and you want to display `name` for each item, the path is `spec.items.name`.
```

> - **widget** - optional widget used to render the field referred to by the `path` property. If you don't provide the widget, a default handler is used depending on the data type provided in the schema. For more information about the available widgets, see [Form widgets](form-widgets.md).

Reviewer: ```suggestion
- **widget** - optional widget used to render the field referred to by the **path** property. If you don't provide the widget, a default handler is used depending on the data type provided in the schema. For more information about the available widgets, see [Form widgets](form-widgets.md).
```

> - **simple** - to display in the simple form. It is `false` by default.

Reviewer: Could you start with the noun / adj+noun as in the previous examples?

> - **advanced** - to display in the advanced form. It is `true` by default.

Reviewer: Could you start with the noun / adj+noun as in the previous examples?

]
```

## list section

The `list` section defines extra columns available in the list. The format is similar to the `form` section, however each entry consists only of two values:

### Item parameters

- **path** - _[required]_ contains the path to the data used for the column.
- **widget** - optional widget used to render the field referred to by the `path` property. By default the value is displayed verbatim. For more information about the available widgets, see [Display widgets](display-widgets.md).

### Example

```json
[{ "path": "spec.url" }, { "path": "spec.priority", "widget": "Badge" }]
```

## details section

The `details` section defines the display structure for the details page. It contains two sections, `header` and `body`, both of which are a list of items to display in the `header` section and the body of the page respectively. The format of the entries is similar to the `form` section, however it has extra options available.

> ### Items parameteres

Reviewer: ```suggestion
### Items parameters
```

]
```

## translations sections

> You can provide this section as a single `translations` section, that contains all available languages, formatted for i18next either as YAML or JSON, based on their paths.

Reviewer: ```suggestion
You can provide this section as a single `translations` section that contains all available languages formatted for i18next either as YAML or JSON, based on their paths.
```

  items: Artikel
```

> If you provde both `translations` and `translations-{lang}` sections, they are merged together.

Reviewer: ```suggestion
If you provide both `translations` and `translations-{lang}` sections, they are merged together.
```
